src/core/defs.jl

```diff
@@ -187,6 +187,24 @@ dim_value_dict(md::ModelDef) = Dict([name => collect(values(dim)) for (name, dim
 Base.haskey(md::ModelDef, name::Symbol) = haskey(md.dimensions, name)
 
 isuniform(md::ModelDef) = md.is_uniform
+
+
+# Helper function invoked when the user resets the time dimension with set_dimension!
+# This function calls set_run_period! on each component definition to reset the first and last values.
+function reset_run_periods!(md, first, last)
+
+    for comp_def in values(md.comp_defs)
```

Comment:
You can use the fn defined in defs.jl:
`compdefs(md::ModelDef) = values(md.comp_defs)`